src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/StartupProjectRegistrar.cs

```diff
@@ -98,6 +83,23 @@ protected override Task DisposeCoreAsync(bool initialized)
         {
             IProjectChangeDescription projectChange = e.Value.ProjectChanges[ConfigurationGeneral.SchemaName];
 
+            if (projectChange.Difference.ChangedProperties.Contains(ConfigurationGeneral.ProjectGuidProperty))
+            {
+                Guid result;
+                if (Guid.TryParse(projectChange.After.Properties[ConfigurationGeneral.ProjectGuidProperty], out result))
+                {
+                    _guid = result;
+                    await AddOrRemoveProjectFromStartupProjectList(initialize: true).ConfigureAwait(false);
+                }
+
+                return;
+            }
+
+            if (_guid == Guid.Empty)
```

Comment:
Instead of persisting the GUID as a field, why don't you *always* read it and just pass it to AddOrRemoveProject - less state to track.